### ISSUE TITLE
feat: Homebrew support

### ISSUE THREAD

**Issue by rudeigerc** [tensorchord/envd#347]
Since `envd` supports both macOS and Linux, it would be better for us to support Homebrew as an alternative choice for installation. `goreleaser` also supports generating and publishing Homebrew Taps [by default](https://goreleaser.com/customization/homebrew/).

References:
- https://github.com/weaveworks/footloose/blob/master/.goreleaser.yml#L29-L39
- https://github.com/weaveworks/homebrew-tap

**Comment by popfido** [comment]
install envd through pip does seems quite strange. Do we need a separate repo to manage the tap?

**Comment by gaocegege** [comment]
@popfido We support pip since data scientists always use it. Thus we build python wheels although envd is a golang project.

Do you prefer `brew install` or `apt install`? I am thinking if we are going in the right direction about installation.

/cc @VoVAllen @kemingy

**Comment by gaocegege** [comment]
> Do we need a separate repo to manage the tap?

I am not familiar with brew. I am a Ubuntu user.

I think the goreleaser can take care of tap and we do not need to maintain them in the repository. Correct me if I am wrong. cc @rudeigerc

**Comment by rudeigerc** [comment]
@popfido @gaocegege 

We do need a tap repository for maintaining Homebrew Formula. goreleaser would handle this for us. It supoorts auto-generating the formula and pushing to the tap repo.

**Comment by gaocegege** [comment]
One of the reasons why I do not like macOS is homebrew :smile:

**Comment by kemingy** [comment]
> One of the reasons why I do not like macOS is homebrew 😄

Same here. I manually install everything on macOS to avoid homebrew breaking the whole system.

BTW, we can also support conda. (not sure about nix 👀

**Comment by gaocegege** [comment]
nix is too hardcore. I do not think we need to support it since our target users may do not even know about it.

**Comment by popfido** [comment]
@gaocegege If in Linux, seldom does a DS use brew to install something, but it is even bad to use pip. Coz it's often a hard job for DS to identify whether the installation is in a system python env or a virtual python or in conda's default env, etc. Placing envd in `/usr/local/bin` by package manager in various system might be more suitable.

**Comment by gaocegege** [comment]
SGTM. It is reasonable.

**Comment by kemingy** [comment]
It's okay if they don't know which `pip` they are using because we don't depend on Python versions. It's just a wrapper to call the binary. Of course, it's better to install with `pipx`.

**Comment by rudeigerc** [comment]
I do understand why you guys don't like Homebrew 😂 However Homebrew is still the mainstream package management system on macOS.

I will have a look at it. It does not quite fit our release process defined in GitHub Actions now.

**Comment by popfido** [comment]
@rudeigerc I really like brew. And I think it is okay for envd to prepare a brew tap. Just a reminder that forcing people using Linux to install an extra package manager(Because brew is not the default package manager for various Linux distribution) is often not necessary.

**Comment by gaocegege** [comment]
@rudeigerc Should I create a new repo for the tape? Could you please suggest a name?

**Comment by popfido** [comment]
> It's okay if they don't know which `pip` they are using because we don't depend on Python versions. It's just a wrapper to call the binary. Of course, it's better to install with `pipx`.

`envd` don't depends on Python version but on the env path. If one installs `envd` into a non-default python env, or he just installs a new default python env to replace the previous one that has installed `envd`, the installation of `envd` might be broken. 

`pipx` seems okay.

**Comment by kemingy** [comment]
> `envd` don't depends on Python version but on the env path. If one installs `envd` into a non-default python env, or he just installs a new default python env to replace the previous one that has installed `envd`, the installation of `envd` might be broken.
> 
> `pipx` seems okay.

Yeah, if we consider the long-term upgrade, a stable package manager would be better.

Good options:

- conda
- pipx
- apt (not sure if we need to support other Linux distributions)

Homebrew is not a good one. But can support it because lots of people are using it.

**Comment by gaocegege** [comment]
Then let's create the homebrew-tap repository.

**Comment by gaocegege** [comment]
@rudeigerc 

You should have the write permission to https://github.com/tensorchord/homebrew-tap .
